network/server/tchannelthrift/node/service.go

```diff
@@ -269,10 +283,33 @@ func (s *service) FetchBlocksRaw(tctx thrift.Context, req *rpc.FetchBlocksRawReq
 	return res, nil
 }
 
+type closeableMetadataResult struct {
```

Comment:
Perhaps place this `closeableMetadataResult` struct and the struct's `OnClose` method below the struct and method implementations for `service` to group by types at a top level within the file?